packages/react-instantsearch-hooks/src/__tests__/useSortBy.test.tsx

```diff
@@ -0,0 +1,36 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import { createInstantSearchTestWrapper } from '../../../../test/utils';
+import { useSortBy } from '../useSortBy';
+
+const items = [
+  { label: 'Featured', value: 'indexName' },
+  { label: 'Price (asc)', value: 'indexName_price_asc' },
+  { label: 'Price (desc)', value: 'indexName_price_desc' },
+];
+
+describe('useSortBy', () => {
+  test('returns the connector render state', async () => {
+    const wrapper = createInstantSearchTestWrapper();
+    const { result, waitForNextUpdate } = renderHook(
+      () => useSortBy({ items }),
+      { wrapper }
+    );
+
+    expect(result.current).toEqual({
+      currentRefinement: 'indexName',
+      options: items,
+      refine: expect.any(Function),
+      hasNoResults: expect.any(Boolean),
+    });
+
+    await waitForNextUpdate();
+
+    expect(result.current).toEqual({
+      currentRefinement: 'indexName',
+      options: items,
+      refine: expect.any(Function),
+      hasNoResults: expect.any(Boolean),
+    });
```

Comment:
Sorry i'm a little late to the party 🥳

I'd love for us to add a test measuring the number of renders, preferably in way that's as close as possible from the real in-app use case.

I'm thinking for example how many re-render will this kind of code trigger, or how does it behave when the parent re-renders too often.

```ts
function MyComponent() {
   useSortBy({ 
         items: [ 
            { label: ... } /* <-- a new reference at each render */ 

         ] /* <-- a new reference at each render */ 
      }  /* <-- a new reference at each render */ 
   )
   ...
   return ...
}
```

@francoischalifour @dhayab

Reply:
`useConnector` [manipulates stable props](https://github.com/algolia/react-instantsearch/blob/6bd156c3977200946e254293ef3d170801bdf6ea/packages/react-instantsearch-hooks/src/useConnector.ts#L19) under the hood so users don't have to think about references. This avoids infinite loops and undesired rerenders.

There's a slight performance cost, but since our Connector props are basic types, it's negligible.